src/components/text-input/text-input.js

```diff
@@ -0,0 +1,78 @@
+import settings from '../../globals/js/settings';
+import mixin from '../../globals/js/misc/mixin';
+import createComponent from '../../globals/js/mixins/create-component';
+import initComponentBySearch from '../../globals/js/mixins/init-component-by-search';
+import handles from '../../globals/js/mixins/handles';
+import eventMatches from '../../globals/js/misc/event-matches';
+import on from '../../globals/js/misc/on';
+
+export default class TextInput extends mixin(createComponent, initComponentBySearch, handles) {
+  /**
+   * Text Input.
+   * @extends CreateComponent
+   * @extends InitComponentBySearch
+   * @extends Handles
+   * @param {HTMLElement} element - The element functioning as a text field.
+   */
+  constructor(element, options) {
+    super(element, options);
+    this.manage(
+      on(this.element, 'click', event => {
+        const toggleVisibilityButton = eventMatches(event, this.options.selectorPasswordVisibilityButton);
+        if (toggleVisibilityButton) {
+          this._toggle({ element, button: toggleVisibilityButton });
+        }
+      })
+    );
+  }
+
+  /**
+   * Toggles the visibility of the password in the input field
+   * @param {Object} obj - The elements that can change in the component
+   * @param {HTMLElement} obj.element - The element functioning as a text field
+   * @param {HTMLElement} obj.button - The button toggling password visibility
+   */
+  _toggle({ element, button }) {
+    /* eslint-disable max-len */
+    const iconVisibilityOff =
+      '<svg width="16" height="11" viewBox="0 0 16 11"><path d="M8 7.5c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 1c-1.7 0-3-1.3-3-3s1.3-3 3-3 3 1.3 3 3-1.3 3-3 3z"></path><path d="M8 10c2.8 0 5.1-1.5 6.9-4.6C13.1 2.5 10.8 1 8 1 5.2 1 3 2.4 1.2 5.4 2.9 8.6 5.2 10 8 10zM8 0c3.3 0 6 1.8 8.1 5.4C14 9.2 11.3 11 8 11S2 9.2 0 5.5C2 1.9 4.6 0 8 0z"></path></svg>';
```

Comment:
Can we toggle `hidden` attribute for those icons instead of changing HTML here?

Reply:
@asudoh I've added another class method `setIconVisibility` to handle changing the `display` attribute rather than changing the markup. Let me know what you think

Reply:
Good try @emyarod! `.style.x` causes usage of inline style (something we avoid) though - So we need to seek for an alternate approach like `hidden` attribute.